packages/buidler-core/src/internal/buidler-evm/provider/filter.ts

```diff
@@ -0,0 +1,130 @@
+import Bloom from "@nomiclabs/ethereumjs-vm/dist/bloom";
+import { BN, bufferToHex, toBuffer } from "ethereumjs-util";
+
+import { RpcLogOutput } from "./output";
+
+export const LATEST_BLOCK = -1;
+
+export enum Type {
+  LOGS_SUBSCRIPTION = 0,
+  PENDING_TRANSACTION_SUBSCRIPTION = 1,
+  BLOCK_SUBSCRIPTION = 2
+}
+
+export interface FilterCriteria {
+  fromBlock: number;
+  toBlock: number;
+  addresses: Buffer[];
+  topics: Array<Array<Buffer | null> | null>;
```

Comment:
I was almost convinced that this was wrong and it should be `Array<Array<Buffer | null> | Buffer | null>` 😅

I think I'd name this `normalizedTopics` or something like that. And use a matching name in `_extractLogTopics`. Maybe even:  `type NormalizedFilterTopics = Array<Array<Buffer | null> | null>`.

Reply:
👍